workers_kv_test.go

```diff
@@ -0,0 +1,369 @@
+			"title": "test_namespace_2"
+			}
+		],
+		"success": true,
+		"errors": [],
+		"messages": [],
+		"result_info": {
+			"page": 1,
+			"per_page": 20,
+			"count": 2,
+			"total_count": 2,
+			"total_pages": 1
+		}
+	}`
+
+	mux.HandleFunc(fmt.Sprintf("/accounts/foo/storage/kv/namespaces"), func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "GET", r.Method, "Expected method 'GET', got %s", r.Method)
+		w.Header().Set("content-type", "application/javascript")
+		fmt.Fprintf(w, response)
+	})
+
+	res, err := client.ListStorageNamespaces(context.Background())
+	want := ListStorageNamespacesResponse{
+		successResponse,
+		[]StorageNamespace{
+			{
+				ID:    "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
+				Title: "test_namespace_1",
+			},
+			{
+				ID:    "yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy",
+				Title: "test_namespace_2",
+			},
+		},
+		ResultInfo{
+			Page:       1,
+			PerPage:    20,
+			Count:      2,
+			TotalPages: 1,
+			Total:      2,
+		},
+	}
+
+	if assert.NoError(t, err) {
+		assert.Equal(t, want.Response, res.Response)
+		assert.Equal(t, want.ResultInfo, res.ResultInfo)
+
+		sort.Slice(res.Result, func(i, j int) bool {
+			return res.Result[i].ID < res.Result[j].ID
+		})
+		sort.Slice(want.Result, func(i, j int) bool {
+			return want.Result[i].ID < want.Result[j].ID
+		})
+		assert.Equal(t, res.Result, want.Result)
+	}
+}
+
+func TestWorkersKV_UpdateStorageNamespace(t *testing.T) {
+	setup(UsingOrganization("foo"))
+	defer teardown()
+
+	namespace := "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
+	response := `{
+		"result": null,
+		"success": true,
+		"errors": [],
+		"messages": []
+	}`
+
+	mux.HandleFunc(fmt.Sprintf("/accounts/foo/storage/kv/namespaces/%s", namespace), func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "PUT", r.Method, "Expected method 'PUT', got %s", r.Method)
+		w.Header().Set("content-type", "application/javascript")
+		fmt.Fprintf(w, response)
+	})
+
+	res, err := client.UpdateStorageNamespace(context.Background(), namespace, &StorageNamespaceRequest{Title: "Namespace"})
+	want := successResponse
+
+	if assert.NoError(t, err) {
+		assert.Equal(t, want, res)
+	}
+}
+
+func TestWorkersKV_CreateStorageKV(t *testing.T) {
+	setup(UsingOrganization("foo"))
+	defer teardown()
+
+	key := "test_key"
+	value := []byte("test_value")
+	namespace := "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
+	response := `{
+		"result": null,
+		"success": true,
+		"errors": [],
+		"messages": []
+	}`
+
+	mux.HandleFunc(fmt.Sprintf("/accounts/foo/storage/kv/namespaces/%s/values/%s", namespace, key), func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "PUT", r.Method, "Expected method 'PUT', got %s", r.Method)
+		w.Header().Set("content-type", "binary/octet-stream")
+		fmt.Fprintf(w, response)
+	})
+
+	res, err := client.CreateStorageKV(context.Background(), namespace, key, bytes.NewReader(value))
+	want := successResponse
+
+	if assert.NoError(t, err) {
+		assert.Equal(t, want, res)
+	}
```

Comment:
i recommend moving `want` definition above the `CreateStorageKV` call. should make reading easier, to be able to see error checking right after call.

Reply:
Will fix.